storage/src/main/java/org/apache/kafka/storage/internals/log/LogConfig.java

```diff
@@ -454,54 +460,101 @@ public static void validateNames(Properties props) {
                 throw new InvalidConfigurationException("Unknown topic config name: " + name);
     }
 
+    /**
+     * Validates the values of the given properties. Can be called by both client and server.
+     * The `props` supplied should contain all the LogConfig properties and the default values are extracted from the
+     * LogConfig class.
+     * @param props The properties to be validated
+     */
     public static void validateValues(Map<?, ?> props) {
         long minCompactionLag = (Long) props.get(TopicConfig.MIN_COMPACTION_LAG_MS_CONFIG);
         long maxCompactionLag = (Long) props.get(TopicConfig.MAX_COMPACTION_LAG_MS_CONFIG);
         if (minCompactionLag > maxCompactionLag) {
             throw new InvalidConfigurationException("conflict topic config setting "
-                + TopicConfig.MIN_COMPACTION_LAG_MS_CONFIG + " (" + minCompactionLag + ") > "
-                + TopicConfig.MAX_COMPACTION_LAG_MS_CONFIG + " (" + maxCompactionLag + ")");
+                    + TopicConfig.MIN_COMPACTION_LAG_MS_CONFIG + " (" + minCompactionLag + ") > "
+                    + TopicConfig.MAX_COMPACTION_LAG_MS_CONFIG + " (" + maxCompactionLag + ")");
         }
+    }
 
-        if (props.containsKey(TopicConfig.REMOTE_LOG_STORAGE_ENABLE_CONFIG)) {
-            boolean isRemoteStorageEnabled = (Boolean) props.get(TopicConfig.REMOTE_LOG_STORAGE_ENABLE_CONFIG);
-            String cleanupPolicy = props.get(TopicConfig.CLEANUP_POLICY_CONFIG).toString().toLowerCase(Locale.getDefault());
-            if (isRemoteStorageEnabled && cleanupPolicy.contains(TopicConfig.CLEANUP_POLICY_COMPACT)) {
-                throw new ConfigException("Remote log storage is unsupported for the compacted topics");
-            }
+    /**
+     * Validates the configured values of the LogConfig. Should be called only by the broker.
+     * The `props` supplied doesn't contain any topic-level configs, only broker-level configs.
+     * The default values should be extracted from the KafkaConfig.
+     * @param props The properties to be validated
+     */
+    public static void validateConfiguredValuesInBroker(Map<?, ?> props) {
```

Comment:
Could you explain more about what's the difference between this method and `validateValuesInBroker` method? I think this method is only invoked when initializing `LogManager` without topic related configs, is that right? From the method name, it's really difficult to identify what they are doing. Thanks.